tools/parse/src/main/scala/org/allenai/nlpstack/parse/PolytreeParser.scala

```diff
@@ -1,31 +1,44 @@
 package org.allenai.nlpstack.parse
 
+import java.io.File
+
 import org.allenai.datastore.Datastores
 import org.allenai.nlpstack.core.DependencyParser
 import org.allenai.nlpstack.core.graph.Graph
 import org.allenai.nlpstack.core.parse.graph.{ DependencyGraph, DependencyNode }
 import org.allenai.nlpstack.core.PostaggedToken
 import org.allenai.nlpstack.parse.poly.polyparser
+import org.allenai.nlpstack.parse.poly.polyparser.{ MultiPolytreeParseSource, ConllX, FileBasedPolytreeParseSource }
 
 /** Wrapper for the polyparser using the DependencyParser interface.
   *
   * @param modelFile filename for the parser model
   * @param modelVersion version of the parser model
   * @param useLocalFile if false, then the model file is found on the datastore
+  * @param cacheFiles a sequence of CoNLL files containing "gold" parses to cache
   */
 class PolytreeParser(
```

Comment:
This should be the main API entry point. It would be great if the jdoc string would describe:

* How do we add parses to the cache?
* How do we update parses in the cache?
* How do we clear the cache?
* How do we list the cache contents?
* Where are the cache stores? I presume datastore, it's not 100% clear from this PR where the datastore API is invoked.